src/main/resources/templates/db_scripts/publication-period/migratePublicationPeriod.sql

```diff
@@ -0,0 +1,127 @@
+CREATE OR REPLACE FUNCTION ${myuniversity}_${mymodule}.migrate_publication_period(
+    jsonb_data jsonb
+) RETURNS jsonb AS $$
+DECLARE
+    pub_period jsonb;
+    start_date text;
+    end_date text;
+    date_type_id text;
+    dates jsonb := '{}'::jsonb;
+BEGIN
+    pub_period := jsonb_data -> 'publicationPeriod';
+    start_date := pub_period ->> 'start';
+    end_date := pub_period ->> 'end';
+
+    -- Determine Date type
+    IF (start_date IS NOT NULL AND end_date IS NOT NULL) THEN
+        date_type_id := '8fa6d067-41ff-4362-96a0-96b16ddce267';
+    ELSIF (start_date IS NOT NULL OR end_date IS NOT NULL) THEN
+        date_type_id := '24a506e8-2a92-4ecc-bd09-ff849321fd5a';
+    ELSE
+        RETURN jsonb_data;
+    END IF;
+
+    -- Build the JSONB Dates object
+    IF start_date IS NOT NULL THEN
+        dates := jsonb_set(dates, '{date1}', to_jsonb(substring(start_date FROM 1 FOR 4)));
+    END IF;
+    IF end_date IS NOT NULL THEN
+        dates := jsonb_set(dates, '{date2}', to_jsonb(substring(end_date FROM 1 FOR 4)));
+    END IF;
+    dates := jsonb_set(dates, '{dateTypeId}', to_jsonb(date_type_id));
```

Comment:
Does it really work?
I've run a similar function and got: `ERROR:  could not determine polymorphic type because input has type unknown`
And another question if there should be `true` as last param:
`jsonb_set(target jsonb, path text[], new_value jsonb [, create_missing boolean])`

Reply:
I tested it on my local machine in Docker and it works correctly both with and without "true" as the last parameter.